lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -39,6 +39,16 @@ def default_priority
     self.priority = MiqAeDomain.highest_priority(tenant) + 1 unless priority
   end
 
+  def lock
+    self.system = true
+    save
```

Comment:
`save!`?  And by extension should the method be `lock!`? The latter is optional..

Also you might want to careful with `lock` as it's part of our extensions to ActiveRecord and is used to acquire a database-level row lock. See https://github.com/ManageIQ/manageiq/blob/master/lib/extensions/ar_lock.rb#L17